src/com/mellanox/cicd/Matrix.groovy

```diff
@@ -1211,7 +1211,7 @@ def buildDocker(image, config) {
     vars += toEnvVars(config, config.env)
 
     withEnv(vars) {
-        if (config.registry_host && image.url.contains(config.registry_host)) {
+        if (config.registry_host && config.registry_auth && image.url.contains(config.registry_host)) {
```

Comment:
but registry_auth is not must param, can be registry with no auth (like public docker repo)

Reply:
yes, you're right. 
Here is PR with a better solution - https://github.com/Mellanox/ci-demo/pull/85